CNF-16438: Add documentation on how to configure CPU architecture (#228)
stolostron/siteconfig: docs/cpu_architecture.md

# Configuring CPU Architecture

There are two places where the CPU architecture can be defined.

> ## Using the ClusterInstance CRD Field

Reviewer: ```suggestion
## Setting a Default Architecture for the Cluster
```

> The default architecture for all the nodes in the cluster can be defined once in the ClusterInstance field `cpuArchitecture`. This is provided to allow easily configuring multiple nodes. It is also valid to use the value `multi` here to show that a mixed architecture deployment is intended.

Reviewer: ```suggestion
The default CPU architecture for all nodes in a cluster can be set using the `cpuArchitecture` field in the ClusterInstance resource. This is provided to allow easily configuring multiple nodes. If a mixed-architecture deployment is required, the value `multi` can be used.
```

If no value is provided here then it will be assumed to be `x86_64`.

```yaml
...
spec:
    cpuArchitecture: multi
...
    nodes:
        - hostname: node1
          cpuArchitecture: aarch64
        ...
        - hostname: node2
          cpuArchitecture: x86_64
        ...
        - hostname: node3
        ...
```

> ## Using the NodeSpec Struct Field

Reviewer: ```suggestion
## Overriding Architecture for Individual Nodes
```

> Each node defined in the Nodes list can have its `cpuArchitecture` explicitly defined.

Reviewer: ```suggestion
Each node in the `nodes` list can specify its `cpuArchitecture` field to override the default set at the cluster level.
```

> If no value is provided here then it will be assumed to be `x86_64` unless a default is provided in the ClusterInstance.

Reviewer: ```suggestion
If no value is provided, the node inherits the cluster-level default (if specified). Otherwise, it defaults to `x86_64`.
```

```yaml
...
nodes:
    - hostname: example-node
      cpuArchitecture: aarch64
      ...
```

> ## ARM Specific Configuration using Assisted Installer

Reviewer: ```suggestion
## Configuring ARM Nodes for Assisted Installer
```

> Installing ARM nodes with Assisted Installer currently requires setting an annotation to ensure the Ironic agent has the correct image. This annotation can be added using the `extraAnnotations` field. This field is valid on both the NodeSpec and ClusterInstance CRD but should only be set for ARM nodes.

Reviewer: ```suggestion
Installing ARM nodes with Assisted Installer currently requires setting an annotation to ensure the Ironic agent has the correct image. This annotation is specified using the `extraAnnotations` field, which is supported at both the cluster and node levels. However, it should only be applied to ARM nodes.
```

```yaml
extraAnnotations:
    InfraEnv:
        infraenv.agent-install.openshift.io/ironic-agent-image-override: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:placeholder
```

> The upstream documentation has more information on this: <https://github.com/openshift/assisted-service/tree/master/docs/hive-integration#ironic-agent-image>.

Reviewer: ```suggestion
For more details, refer to the upstream documentation: <https://github.com/openshift/assisted-service/tree/master/docs/hive-integration#ironic-agent-image>.
```
